arangod/Cluster/Maintenance.cpp

```diff
@@ -1076,13 +1084,25 @@ arangodb::Result arangodb::maintenance::phaseTwo (
 
     // maintenace actions
     report.add(VPackValue("actions"));
-    { VPackObjectBuilder agency(&report);
+    { VPackArrayBuilder agency(&report);
       try {
         std::vector<ActionDescription> actions;
         result = syncReplicatedShardsWithLeaders(
-          plan, cur, local, serverId, actions);
+          plan, cur, local, serverId, feature, actions);
 
-        for (auto const& action : actions) {
+        for (auto& action : actions) {
+          { VPackObjectBuilder b(&report);
+            action.toVelocyPack(report);
+          }
+
+          auto const& name = action.get(NAME);
```

Comment:
spaghetti code.  Everything specific to one type of action needs to happen within that action's class.  See notes on virtualized setState().